pkg/resources/resource_cluster.go

```diff
@@ -339,3 +360,43 @@ func clusterDelete(ctx context.Context, d *schema.ResourceData, meta interface{}
 	}
 	return nil
 }
+
+func clusterImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	metaDb, region, err := utils.GetDBClientFromMeta(meta, d)
+	if err != nil {
+		return nil, err
+	}
+
+	importId := utils.ExtractId(d.Id())
+
+	// Try to fetch by ID first
+	d.Set("use_name_as_id", false)
+	s, err := materialize.ScanCluster(metaDb, importId, false)
+	if err == sql.ErrNoRows {
+		// If not found by ID, try by name
+		s, err = materialize.ScanCluster(metaDb, importId, true)
+		if err != nil {
+			return nil, fmt.Errorf("error importing cluster %s: %s", importId, err)
+		}
+		// If found by name, set use_name_as_id to true
+		d.Set("use_name_as_id", true)
+	} else if err != nil {
+		return nil, fmt.Errorf("error importing cluster %s: %s", importId, err)
+	}
```

Comment:
This looks a bit dangerous! What if someone names a cluster `u7`? This logic will get confused.

What I would recommend is baking the type of ID into the id string itself. Propose we add a prefix to the ID like `ID:` or `NAME:`

```
ID:u7
NAME:name
```

A resource ID that starts with `ID:` is a cluster ID, while a resource ID that starts with `NAME:` is a cluster name.

Then, when someone runs `terraform import <resource ID>`, you can look at the resource ID to determine whether you should look up by name or ID.

This would require a state migration, but I think a pretty straightforward one?